extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/DefaultTenantConfigResolver.java

```diff
@@ -27,18 +26,17 @@
     @Inject
     Instance<TenantConfigResolver> tenantConfigResolver;
 
-    private volatile Map<String, TenantConfigContext> staticTenantsConfig;
-    private volatile TenantConfigContext defaultTenant;
-    private volatile Function<OidcTenantConfig, TenantConfigContext> tenantConfigContextFactory;
-    private volatile Map<String, TenantConfigContext> dynamicTenantsConfig;
+    private final Map<String, TenantConfigContext> dynamicTenantsConfig = new ConcurrentHashMap<>();
```

Comment:
@stuartwdouglas I'm not sure a concurrent map is needed since we still synchronize on it, the sync block is needed to avoid the concurrent connection requests. But I suppose it is performing better anyway, the concurrent map.